src/axom/inlet/inlet_utils.cpp

```diff
@@ -119,14 +119,50 @@ std::string removeBeforeDelimiter(const std::string& path, const char delim)
   return path.substr(pos + 1);
 }
 
-std::pair<int, bool> checkedConvertToInt(const std::string& number)
+std::string removeAllInstances(const std::string& target,
+                               const std::string& substr)
+{
+  std::string result = target;
+  auto pos = result.find(substr);
+  while(pos != std::string::npos)
+  {
+    result.erase(pos, substr.length());
+    pos = result.find(substr);
+  }
+  return result;
+}
+
+bool checkedConvertToInt(const std::string& number, int& result)

```

Comment:
Good feedback - I had been vacillating between the out-params (which Inlet used to use frequently) and the returned pair when I initially wrote this function.

There are a couple other places where I would have found it convenient to use `std::optional`, too.  I think implementing a limited version of the type could be useful (in `inlet_utils` or something) to start, but it might be a good utility type to pull into `core` as well.  Any thoughts on this @white238 @kennyweiss?

Reply:
Yes, I meant output parameters :). I think input parameters are fine.